options/posix/include/sys/select.h

```diff
@@ -16,15 +16,15 @@
 extern "C" {
 #endif
 
-void FD_CLR(int fd, fd_set *);
-int FD_ISSET(int fd, fd_set *);
-void FD_SET(int fd, fd_set *);
-void FD_ZERO(fd_set *);
-
-#define FD_CLR FD_CLR
-#define FD_ISSET FD_ISSET
-#define FD_SET FD_SET
-#define FD_ZERO FD_ZERO
+void __FD_CLR(int fd, fd_set *);
+int __FD_ISSET(int fd, fd_set *);
+void __FD_SET(int fd, fd_set *);
+void __FD_ZERO(fd_set *);
+
+#define FD_CLR(fd, set) __FD_CLR(fd, set)
+#define FD_ISSET(fd, set) __FD_ISSET(fd, set)
+#define FD_SET(fd, set) __FD_SET(fd, set)
+#define FD_ZERO(set) __FD_ZERO(set)
```

Comment:
Why do these need to be macros? Is it not possible to do `#define FD_CLR(fd, set) FD_CLR(fd, set)`?

Reply:
selfref macros would be preferable

Reply:
update: microsoft (and in turn wine) are insane, FD_SET is declared as a type on windows and it's part of the api, so we might have to rename them

I'm open to and would prefer literally any other solution

Reply:
Maybe we could rename FD_xxx to __FD_xxx, and provide the FD_xxx symbols as aliases for ABI backwards compat?

Reply:
If we are going to break ABI for pthread anyway then we might as well get rid of such hacks when we do so

Reply:
I've removed this part of the PR and split it out into #452, that means we can merge this PR now.